### PR TITLE
[TRAFODION-1912]support automatic convert DOS format during bulkload

### PR DESCRIPTION
**Pull request by traflm**
Introduce a new CQD : HIVE_SCAN_SPECIAL_MODE
which is a bitmap, to allow different 'tolerance' suggested by JIRA 1911.
This patch is for DOS format handling. When HIVE_SCAN_SPECIAL_MODE set to 1, HIVE_SCAN will automatically replace \r\n to space\n

Please help to review.

### CODE REVIEW

File: core/sql/executor/ExHdfsScan.cpp [hunk 1]
```diff
@@ -367,6 +367,9 @@ ExWorkProcRetcode ExHdfsScanTcb::work()
   char cursorId[8];
   HdfsFileInfo *hdfo = NULL;
   Lng32 openType = 0;
+  int hiveScanMode = 0;
+
+  hiveScanMode = CmpCommon::getDefaultLong(HIVE_SCAN_SPECIAL_MODE);
```
Comment (selvaganesang) [inline core/sql/executor/ExHdfsScan.cpp]: Usually, CQDs are not accessed directly in executor operators directly. It should be passed as a flag in the TDB. It is possible that the query is compiled in a different process, then this CQD setting won't be available in the executor layer.

Comment (selvaganesang) [inline core/sql/executor/ExHdfsScan.cpp]: Please get this CQD passed as an attribute in TDB

Comment (traflm) [inline core/sql/executor/ExHdfsScan.cpp]: thanks Selva, this explains my puzzle. In the hive regress test, I have to 
cqd CALL_EMBEDDED_ARKCMP 'ON';
It must be due to the reason you mentioned, here, I will add this as a TDB flag.

File: core/sql/executor/ExHdfsScan.h [hunk 2]
```diff
@@ -429,41 +432,92 @@ class ExOrcFastAggrTcb  : public ExOrcScanTcb
...
+   }
+  }
+  else
+  {
+   while (curr < end) {
+     if (*curr == c)
+     {
+         if(count>0 && c == '\n')
+         {
+           if(s[count-1] == '\r') s[count-1] = ' ';
```
Comment (selvaganesang) [inline core/sql/executor/ExHdfsScan.h]: What happens if the column happens to be numeric column.  Is the space character tolerated in the conversion

File: core/sql/executor/ExHdfsScan.h [hunk 3]
```diff
@@ -429,41 +432,92 @@ class ExOrcFastAggrTcb  : public ExOrcScanTcb
...
-       *rdSeen = FALSE;
-       return curr;
+  int count = 0;
+  if( (mode & HIVE_MODE_DOSFORMAT)>0 )  //check outside the while loop to make it faster
+  {
+    while (curr < end) {
+      if (*curr == rd) {
+         if(count>0 && rd == '\n')
+         {
+             if(s[count-1] == '\r') s[count-1] = ' ';
```
Comment (selvaganesang) [inline core/sql/executor/ExHdfsScan.h]: Same comment as above

File: core/sql/executor/ExHdfsScan.h (hunk 2 above)
Comment (traflm) [inline core/sql/executor/ExHdfsScan.h]: I tested with DATE and INT, both works well with extra space. But maybe \0 is better. Let me try it out.

Comment (traflm) [inline core/sql/executor/ExHdfsScan.h]: \0 will not work, I will keep it as space.

**Comment by traflm** [comment]
jenkins, retest

**Comment by traflm** [comment]
If the last column is a NULL, this change will add a space. Need to fix it again...

**Comment by traflm** [comment]
I changed the implementation: not replace the \r with \n, but return back the previous pointer before \r. And in the caller, add that extra position when it needs to move to next field.
Test with INT/DATE/String, all works well, also test with last column as null, works fine too.

**Comment by selvaganesang** [comment]
Change looks good to me. Can you please modify the new test to insert into trafodion table with date as a last column. Also add another test for numeric column as the last column and insert into trafodion table too.

**Comment by traflm** [comment]
Thanks Selva, I will add the test for numeric/date data
